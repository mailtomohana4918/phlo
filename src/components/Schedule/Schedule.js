import React, { useState } from "react";
import Doctor from "../Doctor/Doctor";
import { makeStyles } from "@material-ui/core/styles";
import { Box, TextField, Button, Grid } from "@material-ui/core";
import emailjs from "emailjs-com";
import Header from "../Header/Header";
import Success from "../Success/Success";

const useStyles = makeStyles((theme) => ({
  doctor: {
    width: "280px",
  },
  textBoxContainer: {
    paddingTop: "45px",
    display: "flex",
  },
  boxContainer: {
    paddingTop: "40px",
  },
  butonContainer: {
    paddingTop: "40px",
    paddingLeft: "240px",
  },
  textField: {
    width: "400px",
  },
}));

const bst = new Date().toJSON().slice(0, 10).replace(/-/g, "-");

export default function Schedule() {
  const classes = useStyles();
  const [showModal, setShowModal] = useState(false);
  const [patientDetails, setPatientDetails] = useState({
    name: "",
    emailAddress: "",
    date: bst,
    time: "08:30",
    doctor: "",
  });

  const handleDoctorEntry = (value) => {
    setPatientDetails({
      ...patientDetails,
      doctor: value,
    });
  };

  const updateField = (e) => {
    setPatientDetails({
      ...patientDetails,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
    try{
    await emailjs
      .send(
        "gmail",
        "template_h2ylBlee",
        patientDetails,
        "user_wEOWDQL4XbYZXwuY0BZPC"
      )
      
          setShowModal(true);
      }
      catch(error)
      {
        throw new Error(`Error occurred while sending email ${error}`)
      }
       
  };

  const handleModalClose = () => {
    setShowModal(false);
    setPatientDetails({
      name: "",
      emailAddress: "",
      date: bst,
      time: "08:30",
      doctor: "",
    });
  };

  return (
    <Grid container justify="center">
      <form onSubmit={handleSubmit}>
        <Header />
        <Box className={classes.boxContainer}>
          <Box className={classes.textBoxContainer}>
            <Box className={classes.doctor} mr={2}>
              <Doctor updateDoctorName={handleDoctorEntry} />
            </Box>

            <Box>
              <TextField
                name="name"
                value={patientDetails.name}
                variant="outlined"
                placeholder="Name*"
                onChange={updateField}
                required
                className={classes.textField}
              />
            </Box>
          </Box>
          <Box className={classes.textBoxContainer}>
            <Box mr={2}>
              <TextField
                name="emailAddress"
                value={patientDetails.emailAddress}
                variant="outlined"
                placeholder="Email*"
                type="email"
                onChange={updateField}
                required
                className={classes.textField}
              />
            </Box>

            <Box>
              <TextField
                label="Appointment Date"
                type="date"
                name="date"
                onChange={updateField}
                defaultValue={patientDetails.date}
                InputProps={{ inputProps: { min: patientDetails.date } }}
                InputLabelProps={{
                  shrink: true,
                }}
                required
              />
              <TextField
                label="Time"
                type="time"
                name="time"
                onChange={updateField}
                defaultValue={patientDetails.time}
                InputProps={{ inputProps: { min: "08:30", max: "18:00" } }}
                InputLabelProps={{
                  shrink: true,
                }}
                inputProps={{
                  step: 300,
                }}
                required
              />
            </Box>
          </Box>
          <Box className={classes.butonContainer}>
            <Button type="submit" color="primary" variant="contained" data-testid="schedule-button">
              Schedule
            </Button>
          </Box>
        </Box>
      </form>
      {showModal && <Success handleClose={handleModalClose} />}
    </Grid>
  );
}
