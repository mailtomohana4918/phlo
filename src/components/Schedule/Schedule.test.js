import { render } from "@testing-library/react";
import React from "react";
import Schedule from "./Schedule";

describe("Schedule Component", () => {
  it("renders as expected", () => {
    const { asFragment } = render(<Schedule />);
    expect(asFragment()).toMatchSnapshot();
  });
});
