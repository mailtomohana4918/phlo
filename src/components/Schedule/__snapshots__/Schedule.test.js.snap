// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule Component renders as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <form>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
            >
              Appointment Booking
            </h5>
          </div>
        </div>
      </header>
      <div
        class="MuiBox-root MuiBox-root-183 makeStyles-boxContainer-3"
      >
        <div
          class="MuiBox-root MuiBox-root-184 makeStyles-textBoxContainer-2"
        >
          <div
            class="MuiBox-root MuiBox-root-185 makeStyles-doctor-1"
          >
            <div
              aria-expanded="false"
              class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon"
              role="combobox"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                  for="mui-20710"
                  id="mui-20710-label"
                >
                  Near by doctors
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                >
                  <input
                    aria-autocomplete="list"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    id="mui-20710"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Clear"
                      class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                      tabindex="-1"
                      title="Clear"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-289 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-291"
                    >
                      <span>
                        Near by doctors
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-293"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-textField-5"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  name="name"
                  placeholder="Name"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-289 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-290"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-294 makeStyles-textBoxContainer-2"
        >
          <div
            class="MuiBox-root MuiBox-root-295"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-textField-5"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  name="emailAddress"
                  placeholder="Email"
                  required=""
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-289 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-290"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-296"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
                data-shrink="true"
              >
                Appointment Date
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  min="2020-05-08"
                  name="date"
                  type="date"
                  value="2020-05-08"
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-required Mui-required"
                data-shrink="true"
              >
                Time
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  max="18:00"
                  min="08:30"
                  name="time"
                  type="time"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-311 makeStyles-butonContainer-4"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            data-testid="schedule-button"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Schedule
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
